tests/unit/readers/test_adf15_reader.py

```diff
@@ -0,0 +1,199 @@
+"""Unit tests for the functions in routines.py
+"""
+import unittest
+
+import numpy as np
+from prov.model import ProvEntity
+
+from indica.readers import ADASReader
+
+
+class Testadf15(unittest.TestCase):
+    """Provides unit tests for the reverse() function"""
+
+    def setUp(self):
+        """Setup list for testing"""
+        self.reader = ADASReader()
+        self.element = "ne"
+        self.charge = "9"
+        self.file_type = "pju"
+        self.year = "96"
+
+    def test_read(self):
+        """Checks reading does not give exceptions"""
+        try:
+            _ = self.reader.get_adf15(
+                self.element, self.charge, self.file_type, year=self.year
+            )
+        except Exception as e:
+            raise e
+
+    def test_false_input(self):
+        """Checks reading with wrong inputs raises errors"""
+        element = self.element + ";"
+        try:
+            _ = self.reader.get_adf15(
+                element, self.charge, self.file_type, year=self.year
+            )
+        except Exception as e:
+            assert e
```

Comment:
Can use `.assertRaises` method to check that an `Exception` is raised (as well as checking which to make sure it's the right error you're catching!)

e.g. could replace `try...except` block with: `self.assertRaises(ValueError, self.reader.get_adf15(...))`

Reply:
Tried this but doesn't work.

Reply:
There is a way to get `self.assertRaises()` working but it requires creating another class that inherits from `unittest.TestCase` and defining functions inside that and then calling those functions instead of `try...except`, a bit more work than is necessary I think, especially if this works. See `tests/unit/operators/test_fractional_abundance.py` on my issue #42 branch (it's right at the start).

Reply:
I agree with @gadgil48, expected it to work out of the box but not worth the effort to rewrite since this does the job anyway

Reply:
Why is `unittest.TestCase` being used here? Everywhere else we use pytest, which offers the [raises](https://docs.pytest.org/en/6.2.x/reference.html#pytest.raises) context manager.